src/builtins.ts

```diff
@@ -1638,6 +1640,299 @@ export function compileCall(
       compiler.currentType = Type.void;
       return module.createStore(typeArguments[0].byteSize, arg0, arg1, type.toNativeType(), offset);
     }
+    case "Atomic.load": { // Atomic.load<T!>(offset: usize, constantOffset?: usize) -> *
+      if (operands.length < 1 || operands.length > 2) {
+        if (!(typeArguments && typeArguments.length == 1)) {
+          compiler.error(
+            DiagnosticCode.Expected_0_type_arguments_but_got_1,
+            reportNode.range, "1", typeArguments ? typeArguments.length.toString(10) : "0"
+          );
+        }
+        if (operands.length < 1) {
+          compiler.error(
+            DiagnosticCode.Expected_at_least_0_arguments_but_got_1,
+            reportNode.range, "1", operands.length.toString(10)
+          );
+        } else {
+          compiler.error(
+            DiagnosticCode.Expected_0_arguments_but_got_1,
+            reportNode.range, "2", operands.length.toString(10)
+          );
+        }
+        return module.createUnreachable();
+      }
+      if (!(typeArguments && typeArguments.length == 1)) {
+        if (typeArguments && typeArguments.length) compiler.currentType = typeArguments[0];
+        compiler.error(
```

Comment:
I think you could move on top and unify checking count of type arguments and print error.

Reply:
This will prevent catching further errors. `typeArguments` and operands can be empty.

Reply:
We can definitely improve it. I kept the error checking same style as `load`, `store` implementation.

Reply:
> This will prevent catching further errors. `typeArguments` and operands can be empty.

I mean, you check this twice inside branches but may be better move out and do this only once at first place?
Also 
```ts
if (!(typeArguments && typeArguments.length == 1)) {
   if (typeArguments && typeArguments.length != 0) { ... } // <-- this will be always false
   ...
}
}

Reply:
I saw it. There was a small issue below that error check. Typescript was complaining about `typeArguments` may be undefined, so I left it like that. I will check other possibility to remove duplicate code.